pkg/gardenlet/controller/shoot/shoot_control_migrate.go

```diff
@@ -126,7 +125,8 @@ func (c *Controller) runPrepareShootControlPlaneMigration(o *operation.Operation
 			return nil
 		}),
 		utilerrors.ToExecute("Retrieve the Shoot namespace in the Seed cluster", func() error {
-			if err := botanist.K8sSeedClient.Client().Get(ctx, client.ObjectKey{Name: o.Shoot.SeedNamespace}, namespace); err != nil {
+			botanist.SeedNamespaceObject = &corev1.Namespace{}
+			if err := botanist.K8sSeedClient.Client().Get(ctx, client.ObjectKey{Name: o.Shoot.SeedNamespace}, botanist.SeedNamespaceObject); err != nil {
```

Comment:
```suggestion
			if err := botanist.K8sSeedClient.APIReader().Get(ctx, client.ObjectKey{Name: o.Shoot.SeedNamespace}, botanist.SeedNamespaceObject); err != nil {
```

According to https://github.com/gardener/gardener/blob/b5b0fe3f91d71991d6649518e7d2a12246f18474/pkg/gardenlet/controller/shoot/shoot_control_delete.go#L88-L98 :)

Reply:
If this applied, I guess it should be done also for https://github.com/gardener/gardener/blob/3a394eeb705f1814e27053f4ed618d5dbe95c62d/pkg/gardenlet/controller/shoot/shoot_control_migrate.go#L116

Reply:
Yep, nice catch @vpnachev, thanks!

Reply:
Also noticed that `nil` was returned at the end of the function even if reading the namespace returned an error different than `NotFound`.